packages/mask/src/plugins/Trader/trader/useTradeContext.ts

```diff
@@ -0,0 +1,225 @@
+import { createContext, useMemo } from 'react'
+import { unreachable } from '@dimensiondev/kit'
+import { TradeProvider } from '@masknet/public-api'
+import { getTraderConstants } from '@masknet/web3-shared-evm'
+import {
+    PANCAKESWAP_BASE_AGAINST_TOKENS,
+    PANCAKESWAP_CUSTOM_BASES,
+    QUICKSWAP_BASE_AGAINST_TOKENS,
+    QUICKSWAP_CUSTOM_BASES,
+    SASHIMISWAP_BASE_AGAINST_TOKENS,
+    SASHIMISWAP_CUSTOM_BASES,
+    SUSHISWAP_BASE_AGAINST_TOKENS,
+    SUSHISWAP_CUSTOM_BASES,
+    UNISWAP_BASE_AGAINST_TOKENS,
+    UNISWAP_CUSTOM_BASES,
+    TRADERJOE_BASE_AGAINST_TOKENS,
+    TRADERJOE_CUSTOM_BASES,
+    PANGOLIN_BASE_AGAINST_TOKENS,
+    PANGOLIN_CUSTOM_BASES,
+    WANNASWAP_BASE_AGAINST_TOKENS,
+    WANNASWAP_CUSTOM_BASES,
+    TRISOLARIS_CUSTOM_BASES,
+    TRISOLARIS_BASE_AGAINST_TOKENS,
+    VENONSWAP_CUSTOM_BASES,
+    VENONSWAP_BASE_AGAINST_TOKENS,
+    OPENSWAP_CUSTOM_BASES,
+    OPENSWAP_BASE_AGAINST_TOKENS,
+    MDEX_CUSTOM_BASES,
+    MDEX_BASE_AGAINST_TOKENS,
+} from '../constants'
+import type { TradeContext as TradeContext_ } from '../types'
+import { TargetChainIdContext } from './useTargetChainIdContext'
+
+export const TradeContext = createContext<TradeContext_ | null>(null)
+
+export function useTradeContext(tradeProvider: TradeProvider) {
```

Comment:
What's the difference with `packages/mask/src/plugins/Trader/trader/useGetTradeContext.ts`?

Reply:
this file needs to be removed, it is now useGetTradeContext